test/base/test_history.py

```diff
@@ -716,3 +717,53 @@ def test_trim_history():
             fval_trimmed_man.append(fval_i)
             fval_current = fval_i
     assert fval_trace_trimmed == fval_trimmed_man
+
+
+def test_hd5_history_from_other(history: pypesto.HistoryBase):
+    """Check that we can copy different histories to HDF5 and that the re-loaded history matches the original one."""
+    hdf5_file = tempfile.mkstemp(suffix='.h5')[1]
+    pypesto.Hdf5History.from_history(history, hdf5_file, id_="0")
+
+    # write a second time to test `overwrite` argument
+    with pytest.raises(RuntimeError, match="already exists"):
+        pypesto.Hdf5History.from_history(
+            history, hdf5_file, id_="0", overwrite=False
+        )
+    pypesto.Hdf5History.from_history(
+        history, hdf5_file, id_="0", overwrite=True
+    )
+
+    copied = pypesto.Hdf5History(file=hdf5_file, id="0")
```

Comment:
```suggestion
    copied = pypesto.Hdf5History.from_history(
        history, hdf5_file, id_="0", overwrite=True
    )
```

Reply:
This was intentional to ensure that the file is closed and read anew. Will add a comment.

Reply:
Don't you already ensure this in lines 725 together with 729?

Reply:
Not for the overwrite=True case. But might be sufficient.